page_object/tests/test_account.py

```diff
@@ -0,0 +1,79 @@
+from selenium.webdriver.common.by import By
+from page_object.pages.base_page import BasePage
+from page_object.pages.products_page import Products
+
+
+def test_login(driver, username, password):
+    main_page = BasePage(driver)
+    main_page.enter_username(username)
+    main_page.enter_password(password)
+    main_page.click_login_button()
+    product_page = Products(driver)
+    assert product_page.is_url_matches()
+
+
+def test_login_with_incorrect_username(driver, password):
+    main_page = BasePage(driver)
+    main_page.enter_username("incorrect_username")
+    main_page.enter_password(password)
+    main_page.click_login_button()
+    div_error = driver.find_element(
```

Comment:
actions such as finding an element by a locator shouldn't be in a test function. It should be encapsulated in a page object.